main.go

```diff
@@ -150,6 +150,19 @@ func mustGetProfilePath() string {
 	return filepath.Join(mustGetConfigPath(), globalMinioProfilePath)
 }
 
+func setupProfilingFromEnv(profiler *interface {
+	Stop()
+}) {
+	switch os.Getenv("MINIO_PROFILER") {
```

Comment:
It is simpler to directly move the switch-case code before registerApp() and defer profile.Start(..).Stop()